controllers/user.go

```diff
@@ -0,0 +1,123 @@
+package controllers
+
+import (
+	"encoding/json"
+	"net/http"
+	"regexp"
+	"strconv"
+
+	"github.com/lifeiscontent/webservice/models"
+)
+
+type userController struct {
+	userIDPattern *regexp.Regexp
+}
+
+func (uc userController) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Path == "/users" {
+		switch r.Method {
+		case http.MethodGet:
+			uc.getAll(w, r)
+		case http.MethodPost:
+			uc.post(w, r)
+		default:
+			w.WriteHeader(http.StatusNotImplemented)
+		}
+	} else {
+		matches := uc.userIDPattern.FindStringSubmatch(r.URL.Path)
+		if len(matches) == 0 {
+			w.WriteHeader(http.StatusNotFound)
+		}
+		id, err := strconv.Atoi(matches[1])
```

Comment:
does the `A` in `Atoi` mean `Any`?

Reply:
Atoi is an old term that is used in C (and many other languages) as well, it's just converting a string to an integer. https://golang.org/pkg/strconv/#Atoi

There is also Itoa for converting integer -> string.